### ISSUE TITLE
Overflowing pickle write buffer as set in src/backends/carbon.h

### ISSUE THREAD

**Issue by deanbittner** [github/brubeck#48]
The default setting --
#define PICKLE_BUFFER_SIZE 4096

Is too small, at least for me.  I overflow it.  I've built with a larger buffer, 16384, and no issues after that.  The observed symptom is that all metrics are not visible and present in whisper, Graphite and downstream when I overwrite this buffer.

**Comment by vmg** [comment]
I'm not sure I understand the issue here; the buffer is [checked for capacity](https://github.com/github/brubeck/blob/master/src/backends/carbon.c#L187-L188) before pushing new pickles, so it should not be possible to overflow it. Can you provide a reproduction recipe?

**Comment by deanbittner** [comment]
Yes, I see that now ... I will take a closer look and see if I can get a test case going.
